### PR TITLE
Fix official build pipeline in dnceng

### PR DESCRIPTION
**Pull request by genlu**
@JoeRobich

### CODE REVIEW

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -180,7 +188,7 @@ stages:
       displayName: Publish Assets
       inputs:
         filePath: 'eng\publish-assets.ps1'
-        arguments: '-configuration $(BuildConfiguration) -branchName "$(SourceBranchName)" -gitHubUserName $(Roslyn.GitHubUserName) -gitHubToken $(AccessToken-dotnet-build-bot-public-repo) -gitHubEmail $(Roslyn.GitHubEmail)'
+        arguments: '-configuration $(BuildConfiguration) -branchName "$(SourceBranchName)" -gitHubUserName $(Roslyn.GitHubUserName) -gitHubToken $(AccessToken-dotnet-build-bot-public-repo) -gitHubEmail $(Roslyn.GitHubEmail) -devdivApiKey $(dn-bot-devdiv-packaging-rw)'
```
Comment (JoeRobich) [inline azure-pipelines-official.yml:191]: Should we be using $(_DevDivNugetFeedAccessToken) here?
```suggestion
        arguments: '-configuration $(BuildConfiguration) -branchName "$(SourceBranchName)" -gitHubUserName $(Roslyn.GitHubUserName) -gitHubToken $(AccessToken-dotnet-build-bot-public-repo) -gitHubEmail $(Roslyn.GitHubEmail) -devdivApiKey $(_DevDivNugetFeedAccessToken)'
```

**Review by JoeRobich — APPROVED** [review]

File: azure-pipelines-official.yml (hunk 1 above)
Comment (genlu) [reply azure-pipelines-official.yml]: Yup!

File: azure-pipelines-official.yml [hunk 2]
```diff
@@ -98,19 +106,21 @@ stages:
...
     - task: tagBuildOrRelease@0
       displayName: Tag master validation build
       inputs:
         type: 'Build'
         tags: |
           MasterValidationBuild
-      condition: and(succeeded(), eq(variables['SourceBranchName'], 'master'))
+      condition: and(succeeded(), eq(variables['SourceBranchName'], 'master'), eq(variables['System.TeamProject'], 'DevDiv'))
 
+    # Skip master validation in dnceng until the repo PAT issue is resolved
```
Comment (genlu) [inline azure-pipelines-official.yml:118]: @JoeRobich Skipping master validation in dnceng for now, since "git pull" fails to authenticate at the moment

Comment (RikkiGibson) [reply azure-pipelines-official.yml:118]: Didn't we just make a change that removed some github credentials from a script? Is there any connection between those changes?

For some reason I can't find the PR but I swear I just signed off on it and saw it get merged in the past few days @tmat.

Comment (genlu) [reply azure-pipelines-official.yml]: No, this is accessing roslyn repo in dnceng, not the github one. Anyways, I added another commit to address it

**Comment by genlu** [comment]
@JoeRobich FYI I have made more changes after you signed off

**Review by JoeRobich — APPROVED** [review]

**Comment by JoeRobich** [comment]
@genlu my only thought is that we may want to rebase these changes and target release/dev16.9 since it will be a LTS release.

**Comment by genlu** [comment]
@JoeRobich Makes sense. Will do. Who's approval is required?

**Comment by JoeRobich** [comment]
> Who's approval is required?

No special approvals needed for infrastructure changes.

**Comment by genlu** [comment]
@JoeRobich Done

**Comment by RikkiGibson** [comment]
I'd be interested in reviewing once an official build run of this pipeline is available from dnceng as well as from devdiv.

**Comment by genlu** [comment]
@RikkiGibson I have tested insertion validation change in dnceng, for example:
https://dev.azure.com/dnceng/internal/_build/results?buildId=1012902&view=logs&s=6884a131-87da-5381-61f3-d7acc3b91d76&j=20fcf628-b65c-5865-625a-1cef81cda63b

Testing asset publication change would require changes to publishdata, so I will just let the real run test it. Hopefully we are not expecting another insertion in 16.9 today :)

**Pull request merged**